### PR TITLE
Fixing small problems around counting replies

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing small problems around counting replies related to the new ReplyCount server information.

#### Ticket Link
[MM-17468](https://mattermost.atlassian.net/browse/MM-17468)

### CODE REVIEW

File: src/actions/posts.ts [hunk 1]
```diff
@@ -181,8 +181,13 @@ export function createPost(post: Post, files: any[] = []) {
             pending_post_id: pendingPostId,
             create_at: timestamp,
             update_at: timestamp,
+            reply_count: 0,
         };
 
+        if (post.root_id) {
+            newPost.reply_count = (Selectors.getPostRepliesCount(state, post.root_id) || 0) + 1;
```
Comment (reflog) [inline src/actions/posts.ts:188]: just a thought - maybe worth moving `|| 0` into the selector?

File: src/actions/posts.ts [hunk 2]
```diff
@@ -287,8 +291,13 @@ export function createPostImmediately(post: Post, files: any[] = []) {
             pending_post_id: pendingPostId,
             create_at: timestamp,
             update_at: timestamp,
+            reply_count: 0,
         };
 
+        if (post.root_id) {
+            newPost.reply_count = Selectors.getPostRepliesCount(state, post.root_id) + 1;
```
Comment (reflog) [inline src/actions/posts.ts:298]: no `|| 0` here?

File: src/actions/posts.ts (hunk 1 above)
Comment (jespino) [reply src/actions/posts.ts:188]: good idea

**Pull request merged**